test/test_tp109.py

```diff
@@ -164,59 +167,28 @@ def test_snopt(self):
         self.assertTrue(np.isrealobj(val["con"]))
 
     def test_slsqp(self):
-        self.optimize("slsqp", 1e-7)
+        self.optName = "SLSQP"
+        self.setup_optProb()
+        sol = self.optimize(sens="CS")
+        self.assert_solution(sol, 1e-7)
+        # Check that the function values in the solution are real
+        self.assertTrue(np.isrealobj(sol.objectives["obj"].value))
+        self.assertTrue(np.isrealobj(sol.constraints["con"].value))
 
     def test_autorefine(self):
-        # Optimization Object
-        optProb = Optimization("TP109 Constraint Problem", objfunc)
-
-        # Design Variables (Removed infinite bounds for ALPSO)
-        lower = [0.0, 0.0, -0.55, -0.55, 196, 196, 196, -400, -400]
-        upper = [2000, 2000, 0.55, 0.55, 252, 252, 252, 800, 800]
-        value = [0, 0, 0, 0, 0, 0, 0, 0, 0]
-        optProb.addVarGroup("xvars", 9, lower=lower, upper=upper, value=value)
-
-        # Constraints
-        lower = [0, 0, 0, 0, 0, 0, 0, 0]
-        upper = [None, None, 0, 0, 0, 0, 0, 0]
-        if not USE_LINEAR:
-            lower.extend([0, 0])
-            upper.extend([None, None])
-
-        optProb.addConGroup("con", len(lower), lower=lower, upper=upper)
-
-        # And the 2 linear constriants
-        if USE_LINEAR:
-            jac = np.zeros((1, 9))
-            jac[0, 3] = 1.0
-            jac[0, 2] = -1.0
-            optProb.addConGroup("lin_con", 1, lower=-0.55, upper=0.55, wrt=["xvars"], jac={"xvars": jac}, linear=True)
-
-        # Objective
-        optProb.addObj("obj")
-
-        # Check optimization problem:
-        # optProb.printSparsity()
-
-        # Global Optimizer: ALPSO
-        try:
-            opt1 = OPT("ALPSO")
-        except Error:
-            raise unittest.SkipTest("Optimizer not available:", "ALPSO")
-
+        self.optName = "ALPSO"
+        self.setup_optProb()
         # Get first Solution
-        sol1 = opt1(optProb)
-
-        # Now run the previous solution with SNOPT
-        try:
-            opt2 = OPT("SNOPT")
-        except Error:
-            raise unittest.SkipTest("Optimizer not available:", "SNOPT")
+        sol1 = self.optimize(sens="CS")
 
-        sol2 = opt2(sol1)
+        # Now re-optimize with SNOPT
```

Comment:
Yes, we run ALPSO to get near the optimum, then use SNOPT to "refine" by cold starting from the best ALPSO point. This PR does not change any of the tests, this is something that has been there for ages, and it's good because it tests that the `Solution` object can be used to directly cold start optimizations. Although we should check that the DVs get transferred I guess..